logs/parse.go

```diff
@@ -69,7 +69,7 @@ var EscapeMatchers = map[rune]PrefixEscape{
 	},
 	// User name
 	'u': {
-		Regexp: `.+?`,
+		Regexp: `(?:[^ ]|[^ ]{2,})?`,
```

Comment:
This comment applies both here and to the "Database name" change below:

```suggestion
		Regexp: `.*?`,
```

So, the original regexp here is a non-greedy match of one or more characters, because identifiers like user cannot be empty but (when quoted) can contain any character (up to the max identifier length, 63 bytes without recompiling Postgres). (The current bug is that while usernames cannot be empty, they can be logged as empty when logging from a non-session process, because there's no associated user.)

Your regexp here is an optional match of either a non-space, or two or more non-spaces. That will capture most valid usernames, and since it's an optional match, it will also match an empty string (fixing the bug). But it will also reject user names including spaces, or consisting entirely of spaces (neither of these is probably a good idea, but we should support it if possible).

I'm proposing that instead, we change the regexp to capture zero or more characters, instead of one or more, and still keep it as a non-greedy match. This passes the tests, including the test you've added, and I think it should also allow valid user names that have spaces.

Can you confirm if this works for you?